src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +413,40 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, keys, val):
+    """Add target information to run_info under target_data.
+
+    Will Recursively construct a nested dict with the keys provided.
```

Comment:
OK, but we have no current use for this, and this is not as straightforward as it appears. E.g., if you call:

`report_target_info(scope, target, ['this', 'is', 'a', 'deeply', 'namespaced'], 'value1')`
and then
`report_target_info(scope, target, ['this', 'is', 'another', 'deeply', 'namespaced'], 'value2')`

You might reasonably expect the value for (scope, target) to be:
```
{ 'this': { 'is' : { 
  'a': {'deeply': {'namespaced': 'value1' }},
  'another': {'deeply': {'namespaced': 'value2' }}
}}}
```

But actually, if I understand the current code correctly, it would be:

```
{ 'this': { 'is' : { 
  'another: {'deeply': {'namespaced': 'value2' }}
}}}
```

So the semantics are unclear, and it might be best to just avoid the problem entirely, rather than anticipate usage that hasn't actually been needed yet?